### PR TITLE
Update Boost to 1.81

### PR DESCRIPTION
**Pull request by AtariDreams**
I will be the first to say it; I did in fact contribute to this release of Boost. However, it fixes a platitude of issues as this is a bugfix release.

Therefore, I think updating to 1.81 is worth it.

### CODE REVIEW

**Comment by frederick-vs-ja** [comment]
This assertion failed, but I failed to reproduce the error on Godbolt...

https://github.com/microsoft/STL/blob/8ddf4da23939b5c65587ed05f783ff39b8801e0f/tests/std/tests/LWG3234_math_special_overloads/test.cpp#L200

Perhaps we can (temporarily) change this line as indicated (`2` might be replaced with some suitable relative error)

```C++
assert(expect_epsilons(comp_ellint_2(0), acos(-1.0) / 2, 2));
```

**Comment by fsb4000** [comment]
You also need to update this commit hash: https://github.com/microsoft/STL/blob/8ddf4da23939b5c65587ed05f783ff39b8801e0f/docs/cgmanifest.json#L8-L9

**Comment by AtariDreams** [comment]
Done!

**Review by fsb4000 — APPROVED** [review]

**Comment by fsb4000** [comment]
>Perhaps we can (**temporarily**) change this line

@frederick-vs-ja  Do we need to create an issue about this?

**Comment by frederick-vs-ja** [comment]
> > Perhaps we can (**temporarily**) change this line
> 
> @frederick-vs-ja Do we need to create an issue about this (here or maybe in boost-math repo)?

I think I've reproduced the error, so I reported the issue in boost-math repo.

File: tests/std/tests/LWG3234_math_special_overloads/test.cpp [hunk 1]
```diff
@@ -197,7 +197,7 @@ void test_comp_ellint_2() {
     static_assert(is_same_v<decltype(comp_ellint_2f(0)), float>);
     static_assert(is_same_v<decltype(comp_ellint_2l(0)), long double>);
     static_assert(is_same_v<decltype(comp_ellint_2(Ambiguous{})), Ambiguous>);
-    assert(comp_ellint_2(0) == acos(-1.0) / 2);
+    assert(expect_epsilons(comp_ellint_2(0), acos(-1.0) / 2, 2));
```
Comment (CaseyCarter) [inline tests/std/tests/LWG3234_math_special_overloads/test.cpp:200]: For other reviewers: As discussed in https://github.com/boostorg/math/issues/909, this is an expected behavioral change due to re-implementation of the elliptical integrals to greatly increase performance for common cases at the expense of a bit of error for some special cases like π / 2. (No change requested.)

**Review by CaseyCarter — APPROVED** [review]
FYI, I've verified that the commit hash corresponds to the boost-1.81.0 tag in the boost-math repo (https://github.com/boostorg/math/tags).

File: tests/std/tests/LWG3234_math_special_overloads/test.cpp (hunk 1 above)
Comment (strega-nil-ms) [reply tests/std/tests/LWG3234_math_special_overloads/test.cpp:200]: Could we instead use the `isclose` function from #3077? I really dislike the use of epsilons here.

**Review by strega-nil-ms — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

**Comment by fsb4000** [comment]
```
Error LNK2019: unresolved external symbol "void __cdecl std::swap<double,0>(double &,double &)" (??$swap@N$0A@@std@@YAXAEAN0@Z) referenced in function "bool __cdecl isclose<double>(double,double,int)" (??$isclose@N@@YA_NNNH@Z)
```

Adding `#include <utility>` fixed the test.

**Review by CaseyCarter — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for updating the Boost.Math dependency! :white_check_mark: :tada: :smile_cat:

**Pull request merged**